integration/nemo/examples/supervised_fine_tuning/README.md

```diff
@@ -0,0 +1,190 @@
+
+#### 1. Download the datasets
+We download the datasets from HuggingFace:
+```
+mkdir Data
+cd Data
+git clone https://huggingface.co/datasets/tatsu-lab/alpaca
+git clone https://huggingface.co/datasets/databricks/databricks-dolly-15k
+git clone https://huggingface.co/datasets/OpenAssistant/oasst1
+cd ..
+```
+
+#### 2. Preprocess each dataset
+We preprocess each dataset using the following script:
+``` 
+mkdir Data/Processed
+mkdir Data/Processed/alpaca
+python utils/preprocess_alpaca.py --training_file Data/alpaca/data/train-00000-of-00001-a09b74b3ef9c3b56.parquet --output_dir Data/Processed/alpaca
+mkdir Data/Processed/dolly
+python utils/preprocess_dolly.py --training_file Data/databricks-dolly-15k/databricks-dolly-15k.jsonl --output_dir Data/Processed/dolly
+mkdir Data/Processed/oasst1
+python utils/preprocess_oasst1.py --training_file Data/oasst1/data/train-00000-of-00001-b42a775f407cee45.parquet --validation_file Data/oasst1/data/validation-00000-of-00001-134b8fd0c89408b6.parquet --output_dir Data/Processed/oasst1
+```
+
+#### 3. Combine for centralized training
+We also generate a combined version for centralized training baseline:
+```
+mkdir Data/Processed/combined
+python utils/combine_jsonl.py --file_list Data/Processed/alpaca/training.jsonl Data/Processed/dolly/training.jsonl Data/Processed/oasst1/training.jsonl --output_path Data/Processed/combined/training.jsonl
+python utils/combine_jsonl.py --file_list Data/Processed/alpaca/validation.jsonl Data/Processed/dolly/validation.jsonl Data/Processed/oasst1/validation.jsonl --output_path Data/Processed/combined/validation.jsonl
+python utils/combine_jsonl.py --file_list Data/Processed/alpaca/testing.jsonl Data/Processed/dolly/testing.jsonl Data/Processed/oasst1/testing.jsonl --output_path Data/Processed/combined/testing.jsonl
+```
+
+## Federated learning simulations
+Next, we are using NVFlare's [POC mode](https://nvflare.readthedocs.io/en/main/getting_started.html#setting-up-poc) to simulate 
+each client training on their own dataset locally, centralized training with all three dataset combined, and all three clients training together using the 
+[FedAvg](https://arxiv.org/abs/1602.05629) algorithm implemented in NVFlare.
+
+#### 1. Local and Centralized SFT
+For single-site and centralized training experiments, we create the poc workspaces:
+```
+nvflare poc --prepare -n 1
+```
+For 1.3B model experiment, we start the NVFlare system with one GPU:
+```
+nvflare poc --start --gpu 0
+```
+For better usability, open a new terminal and start the [admin command prompt](https://nvflare.readthedocs.io/en/main/real_world_fl/operation.html#admin-command-prompt):
+```
+nvflare poc --start --package admin
+```
+
+We create the configuration files and modify them to include the current directory path to access the dataset and pre-trained LLM.
+At this point, we also modify the data path and local number of clients.
+
+In a standard terminal, run
```

Comment:
This should come before `nvflare poc --start --package admin@...`

Reply:
updated